### PR TITLE
Issue #7895: minor-addRedundantThis

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Fixes: #7895 
major: (https://github.com/checkstyle/checkstyle/pull/11053)

### CODE REVIEW

**Comment by Rahulkhinchi03** [comment]
I think this need to be merged first in fixing (https://github.com/checkstyle/checkstyle/pull/11053)

**Comment by Rahulkhinchi03** [comment]
`Caused by: java.lang.ClassCastException: class com.puppycrawl.tools.checkstyle.checks.coding.AbstractFrameCheck$MethodFrame cannot be cast to class com.puppycrawl.tools.checkstyle.checks.coding.AbstractFrameCheck$ClassFrame (com.puppycrawl.tools.checkstyle.checks.coding.AbstractFrameCheck$MethodFrame and com.puppycrawl.tools.checkstyle.checks.coding.AbstractFrameCheck$ClassFrame are in unnamed module of loader 'app')
`

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java [hunk 1]
```diff
@@ -265,60 +259,66 @@
...
+     */
     private boolean checkFields = true;
-    /** Control whether to check references to methods. */
+    /**
+     * Control whether to check references to methods.
+     */
     private boolean checkMethods = true;
-    /** Control whether to check only overlapping by variables or arguments. */
+    /**
+     * Control whether to check only overlapping by variables or arguments.
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java:320]: If text isn't being changed, why do we need to reformat all these javadocs? These should be reverted.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java [hunk 2]
```diff
@@ -22,9 +22,7 @@
 import static com.google.common.truth.Truth.assertWithMessage;
 import static com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck.MSG_METHOD;
 import static com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck.MSG_VARIABLE;
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertNotNull;
-import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertTrue;
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java:25]: Bad change.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/AbstractFrameCheck.java [hunk 3]
```diff
@@ -0,0 +1,751 @@
...
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CheckUtil;
+import com.puppycrawl.tools.checkstyle.utils.ScopeUtil;
+
+/**
+ * Abstract class for checks which require to work with and identify many frames.
+ */
+@FileStatefulCheck
+public abstract class AbstractFrameCheck extends AbstractCheck {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/AbstractFrameCheck.java:42]: Should any of the RequireThis tests be moved to `AbstractFrameCheckTest`? Are any doing direct reflection on things here?

**Review by rnveach — CHANGES_REQUESTED** [review]
@Rahulkhinchi03 You will want to look over the assertions changes you are making. Quite a few of your changes are undoing my work in https://github.com/checkstyle/checkstyle/issues/9142 making me think you are not working with the latest master when you do your work. I am not saying you need to convert everything old to Truth, but you can't be undoing the changes I have made.

In addition, with Truth you can't end assertions with `that(...)`. You must conclude it what you are asserting like `isTrue` or `isEqualTo` or etc....

In addition, a lot of changes do not look to have any purpose. Those type of changes should be reverted if you aren't adding new code, new words to the javadoc, etc... It just creates extra noise in reviews.

**Comment by rnveach** [comment]
> How I can resolve this?

Please point to test that is failing with that.

**Review by rnveach — CHANGES_REQUESTED** [review]
Why shouldn't this work go under Issue https://github.com/checkstyle/checkstyle/issues/8126 ?

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java (hunk 2 above)
Comment (Rahulkhinchi03) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java:25]: > Why shouldn't this work go under Issue #8126?

So Should I refer my changes to that PR?

The abstract frame check is used for both RedundantThis and RequireThis and This was the reason of Code coverage as the frames were not involved in Redundant This.

**Comment by Rahulkhinchi03** [comment]
> Why shouldn't this work go under Issue #8126 ?

So Should I refer my changes to that PR?

The abstract frame check is used for both RedundantThis and RequireThis and This was the reason of Code coverage as the frames were not involved in Redundant This.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java [hunk 4]
```diff
@@ -163,17 +157,19 @@ public void testGithubIssue41() throws Exception {
...
         final RequireThisCheck check = new RequireThisCheck();
-        assertNotNull(check.getAcceptableTokens(), "Acceptable tokens should not be null");
-        assertNotNull(check.getDefaultTokens(), "Acceptable tokens should not be null");
-        assertNotNull(check.getRequiredTokens(), "Acceptable tokens should not be null");
+        assertWithMessage("Acceptable tokens should not be null")
+                .that(check.getAcceptableTokens());
+        assertWithMessage("Acceptable tokens should not be null")
+                .that(check.getDefaultTokens());
+        assertWithMessage("Acceptable tokens should not be null")
+                .that(check.getRequiredTokens());
```
[the remark this answers is not in the document]
Comment (Rahulkhinchi03) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java:165]: @rnveach could you please provide me the documentation for learning Truth and when to change and when to not to learn more about it?

**Comment by Rahulkhinchi03** [comment]
> > How I can resolve this?
> 
> Please point to the test that is failing with that.

There is a total of 4 classes that are getting the same error.

Here is one:
```
    @Test
    public void testWithAnonymousClass() throws Exception {
        final String[] expected = {
            "28:25: " + getCheckMessage(MSG_METHOD, "doSideEffect", ""),
            "57:17: " + getCheckMessage(MSG_VARIABLE, "foobar", ""),
        };
        verifyWithInlineConfigParser(
                getPath("InputRequireThisAnonymousEmpty.java"),
                expected);
    }
```

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/AbstractFrameCheck.java (hunk 3 above)
Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/AbstractFrameCheck.java:42]: No. Any RequireThis test is not moved to AbstractFrameCheck.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (rnveach) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java:165]: If there is nothing in the issue associated to it, then google is your best friend. I am learning Truth on the go myself.

**Comment by rnveach** [comment]
>     Why shouldn't this work go under Issue #8126 ?
> So Should I refer my changes to that PR?

Issue, not PR, but yes this is my question. If what your doing matches what the issue is saying, then it should be tied to that. This is not really a minor change as your PR title states.

**Comment by rnveach** [comment]
> How I can resolve this?

````
Caused by: java.lang.ClassCastException: com.puppycrawl.tools.checkstyle.checks.coding.AbstractFrameCheck$MethodFrame cannot be cast to com.puppycrawl.tools.checkstyle.checks.coding.AbstractFrameCheck$ClassFrame
	at com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck.collectDeclarations(RequireThisCheck.java:553)
	at com.puppycrawl.tools.checkstyle.checks.coding.AbstractFrameCheck.beginTree(AbstractFrameCheck.java:111)
````

Was the necessary code copied pretty much exactly as is during the move? To me this sounds like something critical is missing.

**Comment by rnveach** [comment]
@Rahulkhinchi03 Please keep this PR to only 1 commit. No merge commits.

**Comment by Rahulkhinchi03** [comment]
Github, rebase

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java [hunk 5]
```diff
@@ -463,13 +463,10 @@ public void testClearState() throws Exception {
...
 
-        assertWithMessage("Ast should contain CLASS_DEF")
-                .that(classDef.isPresent())
-                .isTrue();
-        assertWithMessage("State is not cleared on beginTree")
-                .that(TestUtil.isStatefulFieldClearedDuringBeginTree(check, classDef.get(),
-                        "current", current -> ((Collection<?>) current).isEmpty()))
-                .isTrue();
+        assertWithMessage("Ast should contain CLASS_DEF").that(classDef.isPresent());
+        assertTrue(
```
[the remark this answers is not in the document]
Comment (Rahulkhinchi03) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java]: Done

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (Rahulkhinchi03) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java (hunk 1 above)
Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java]: Done

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java (hunk 2 above)
Comment (Rahulkhinchi03) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java:26]: Done

**Comment by Rahulkhinchi03** [comment]
> > How I can resolve this?
> 
> ```
> Caused by: java.lang.ClassCastException: com.puppycrawl.tools.checkstyle.checks.coding.AbstractFrameCheck$MethodFrame cannot be cast to com.puppycrawl.tools.checkstyle.checks.coding.AbstractFrameCheck$ClassFrame
> 	at com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck.collectDeclarations(RequireThisCheck.java:553)
> 	at com.puppycrawl.tools.checkstyle.checks.coding.AbstractFrameCheck.beginTree(AbstractFrameCheck.java:111)
> ```
>
> Was the necessary code copied pretty much exactly as is during the move? To me this sounds like something critical is missing.

So, how to resolve now?

**Comment by Rahulkhinchi03** [comment]
I think I should close this PR and create a new PR as I think I created a mess in this PR?

**Pull request closed without merging**